src/adapters/book-adapters.ts

```diff
@@ -0,0 +1,245 @@
+            latitude: number,
+            longitude: number,
+        },
+    status: string,
+    description: string,
+    genre: string,
+    author: string,
+    createdAt: string,
+    updatedAt: string,
+    __v: number,
+    imageURL?: string
+}
+/**
+ * This function will get all books for users with the provided information
+ * @param {String} queryBook - The query object containing information for query parameters.
+ * @param {String} queryBook.title - Book's title as query parameter.
+ * @param {String} queryBook.author - Book's author as query parameter.
+ * @param {String} queryBook.sort - Sort as query parameter.
+ * @param {String} queryBook.fields - Fields as query parameter.
+ * @param {Number} queryBook.searchRadius - Search radius as query parameter.
+ * @param {Number} queryBook.latitude - Latitude as query parameter.
+ * @param {Number} queryBook.longitude - Longitude as query parameter.
+ * @example
+ * const queryBook = {
+ *  title = "Pachinko",
+ *  author = "Min Jin Lee",
+ *  sort = "CreatedAt",
+ *  fields = "Genre",
+ *  searchRadius = 25,
+ *  latitude = 20,
+ *  longitude = 40
+ * };
+ * export const getAllBooksAdapter = async(queryBook) => {
+ * // function implementation here
+ * };
+ * @returns {Promise<Object>} A promise that resolves get all books(for users) information.
+ */
+// Get all books for users
+export const getAllBooksAdapter = async(queryBook:queryBook | undefined):Promise<booksSchema[]> => {
+   let url = 'http://localhost:8000/api/v1/books'
+    if(queryBook){
+        const queryParams = Object.entries(queryBook).map((bookFields)=>{
+            const key = bookFields[0]
+            const value = bookFields[1]
+            return `${key}=${value}`
+        }).join('&') 
+       
+        url = `http://localhost:8000/api/v1/books?${queryParams}`
+    }
+    const data = await fetchAPIData(url, 'GET', undefined)
+    return data
+}
+
+/**
+ * Get all books for owner based on the authentication route.
+ * @example
+ * export const getAllBooksOwnerAdapter = async() =>{
+ * // function implementation here 
+ * };
+ * @returns {Promise<Object>} A promise that resolves get all books (for owner) information.
+ */
+// Get all books for owner 
+export const getAllBooksOwnerAdapter = async():Promise<booksSchema[]> => {
+    const url = 'http://localhost:8000/api/v1/books/user'
+    const data = await fetchAPIData(url, 'GET', undefined)
+    return data
+}
+
+// Get single book, delete, and update adapter
+/**
+ * This function will get a single book with the provided information
+ * @param {String} bookId - Id of the book.
+ * @example
+ * const bookId = "642269aef562cad511ed0a73" 
+ * export const getSingleBookAdapter = async(bookId)=>{
+ * // function implementation here
+ * };
+ * @returns {Promise<Object>} A promise that resolves get single book information.
+ */
+// get single book adapter
+export const getSingleBookAdapter = async(bookId:string): Promise<booksSchema[]>=>{
```

Comment:
Lara, I think this one is not returning an array. right? booksSchema[]?

Reply:
I think this one returns an array. If 1 owner has several books listed, it will return as an array right?